packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -175,87 +171,83 @@ export class HttpPlugin extends BasePlugin<Http> {
   }
 
   /**
-   * Injects span's context to header for distributed tracing and finishes the
-   * span when the response is finished.
+   * Attach event listeners to a client request to end span and add span attributes.
+   *
    * @param request The original request object.
    * @param options The arguments to the original function.
    * @param span representing the current operation
    */
-  private _getMakeRequestTraceFunction(
+  private _traceClientRequest(
     request: ClientRequest,
     options: ParsedRequestOptions,
     span: Span
-  ): Func<ClientRequest> {
-    return (): ClientRequest => {
-      this._logger.debug('makeRequestTrace by injecting context into header');
-
-      const hostname =
-        options.hostname ||
-        options.host?.replace(/^(.*)(\:[0-9]{1,5})/, '$1') ||
-        'localhost';
-      const attributes = utils.getOutgoingRequestAttributes(options, {
-        component: this.component,
-        hostname,
-      });
-      span.setAttributes(attributes);
-
-      request.on(
-        'response',
-        (response: IncomingMessage & { aborted?: boolean }) => {
-          const attributes = utils.getOutgoingRequestAttributesOnResponse(
-            response,
-            { hostname }
-          );
-          span.setAttributes(attributes);
-
-          this._tracer.bind(response);
-          this._logger.debug('outgoingRequest on response()');
-          response.on('end', () => {
-            this._logger.debug('outgoingRequest on end()');
-            let status: Status;
-
-            if (response.aborted && !response.complete) {
-              status = { code: CanonicalCode.ABORTED };
-            } else {
-              status = utils.parseResponseStatus(response.statusCode!);
-            }
```

Comment:
Why there is so much changes detected is it because the gts package has been updated ?

Reply:
Mostly it's an indentation change. It used to return a function which was immediately bound to a span and then called which returned a client request (`return type: Func<ClientRequest>`). Now it just returns the client request instead of the function because context is already active.

Reply:
in a drastically simplified form, this:

```typescript
function getMakeClientRequestFunction() {
  return function () {
    return new ClientRequest();
  }
}

const fn = getMakeClientRequestFunction();
tracer.bind(fn, span);
const request = fn();
```

has now become

```typescript
function getClientRequest() {
  return new ClientRequest();
}

tracer.withSpan(span, () => {
  getClientRequest();
})
```

Reply:
thx